src/RestApi/StoreApi/Controllers/CartShippingRates.php

```diff
@@ -75,50 +75,22 @@ public function register_routes() {
 	 * @return \WP_Error|\WP_REST_Response
 	 */
 	public function get_items( $request ) {
+		if ( ! wc_shipping_enabled() ) {
+			return new RestError( 'woocommerce_rest_shipping_disabled', __( 'Shipping is disabled.', 'woo-gutenberg-products-block' ), array( 'status' => 404 ) );
+		}
+
 		$controller = new CartController();
 		$cart       = $controller->get_cart_instance();
 
 		if ( ! $cart || ! $cart instanceof \WC_Cart ) {
 			return new RestError( 'woocommerce_rest_cart_error', __( 'Unable to retrieve cart.', 'woo-gutenberg-products-block' ), array( 'status' => 500 ) );
 		}
 
-		if ( ! empty( $request['country'] ) ) {
-			$valid_countries = WC()->countries->get_shipping_countries();
-
-			if (
-				is_array( $valid_countries ) &&
-				count( $valid_countries ) > 0 &&
-				! array_key_exists( $request['country'], $valid_countries )
-			) {
-				return new RestError(
-					'woocommerce_rest_cart_shipping_rates_invalid_country',
-					sprintf(
-						/* translators: 1: valid country codes */
-						__( 'Destination country code is not valid. Please enter one of the following: %s', 'woo-gutenberg-products-block' ),
-						implode( ', ', array_keys( $valid_countries ) )
-					),
-					[ 'status' => 400 ]
-				);
-			}
-		}
-
-		$request = $this->validate_shipping_address( $request );
-
-		if ( is_wp_error( $request ) ) {
-			return $request;
-		}
-
-		$cart_items = $controller->get_cart_items(
-			function( $item ) {
-				return ! empty( $item['data'] ) && $item['data']->needs_shipping();
-			}
-		);
-
-		if ( empty( $cart_items ) ) {
+		if ( ! $cart->needs_shipping() ) {
 			return rest_ensure_response( [] );
 		}
 
-		$packages = $this->get_shipping_packages( $request );
+		$packages = $controller->get_shipping_packages();
```

Comment:
We’re asking for shipping information without providing any address, currently, the address is still passed but is not saved to the session, and this method is reaching to the session to get this non-existing data

Reply:
Why is it passed? I removed the address passing from this endpoint so it just uses session. The cart/update-shipping endpoint does accept address data.

Reply:
we didn’t update the js hooks endpoint to use the new endpoints, I will do it